src/Dumplie/Application/Query/Customer/Result/Cart.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+declare (strict_types = 1);
+
+namespace Dumplie\Application\Query\Customer\Result;
+
+use Dumplie\Application\Exception\InvalidArgumentException;
+
+final class Cart
+{
+    /**
+     * @var CartItem[]|array
+     */
+    private $items;
+
+    /**
+     * @var float
+     */
+    private $totalPrice;
+
+    /**
+     * @param array|CartItem[] $items
+     * @throws InvalidArgumentException
+     */
+    public function __construct(array $items = [])
```

Comment:
Maybe instead of enforcing currency in constructor just ensure that all products are in same currency?
Eg. first added item sets private currency property and all later added must have same currency.